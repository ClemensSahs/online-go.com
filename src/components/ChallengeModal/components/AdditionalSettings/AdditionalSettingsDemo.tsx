import * as React from 'react';
import {
  _
} from "translate";

import AdditionalSettingsHeaderDemo from './AdditionalSettingsHeaderDemo';
import AdditionalSettingsBoardSize from './AdditionalSettingsBoardSize';

import {
  ChallengeModes,
  AdditionalSettingsDemoProps
} from '../../types';


const AdditionalSettingsDemo = ({
  challenge,
  conf,
  forking_game,
  mode,
  ranked,
  rules,
  update_demo_rules,
  update_board_size,
  update_board_width,
  update_board_height,
}: AdditionalSettingsDemoProps) => {

  let enable_custom_board_sizes = mode === ChallengeModes.DEMO || !ranked;

  return (
    <div id="challenge-basic-settings" className="right-pane pane form-horizontal" role="form">
      <AdditionalSettingsHeaderDemo rules={rules} update_demo_rules={update_demo_rules} />

      {!forking_game &&
        <AdditionalSettingsBoardSize
          enable_custom_board_sizes={enable_custom_board_sizes}
          selected_board_size={conf.selected_board_size}
          board_width={challenge.game.width}
          board_height={challenge.game.height}
          update_board_size={update_board_size}
          update_board_width={update_board_width}
          update_board_height={update_board_height}
        />
      }


    </div>
  );
};


export default AdditionalSettingsDemo;
