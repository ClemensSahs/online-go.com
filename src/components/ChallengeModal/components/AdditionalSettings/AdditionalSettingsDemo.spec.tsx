
import * as React from 'react';
import * as data from "data";

import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import AdditionalSettingsDemo from './AdditionalSettingsDemo';
import {
  componentSanityChecks,
  runPropsMatrix,
  testCheckBox,
} from 'testUtility';
import {
  ChallengeModes
} from '../../types';

jest.mock('data');
// jest.mock('data', () => ({
//     get: jest.fn((x: string, defaultValue) => {
//       console.log('mock '+ x);
//       if (x === 'config.aga_rankings_enabled') {
//         return true;
//       }
//       return defaultValue;
//     })
// }));

jest.mock('translate', () => ({
    _: jest.fn(x => x),
}));

const getBaseProps = (customProps = {}) => {
  const baseProps = {
    challenge: {
      game: {
        width: 19,
        height: 19
      }
    },
    conf: {
      selected_board_size: '19x19'
    },
    forking_game: '',
    mode: ChallengeModes.DEMO,
    ranked: false,
    update_demo_rules: (ev) => console.log(),
    update_board_size: (ev) => console.log(),
    update_board_width: (ev) => console.log(),
    update_board_height: (ev) => console.log(),
    update_ranked: (ev) => console.log(),
    update_aga_ranked: (ev) => console.log(),
  };
  return {
    ...baseProps,
    ...customProps
  };
}


runPropsMatrix([
  ['normal', getBaseProps()],
  ['as demo', getBaseProps({
    mode: ChallengeModes.DEMO
  })],
],(labelSubfix: string, props) => {
  componentSanityChecks(`BasicSettings.${labelSubfix}`, <AdditionalSettingsDemo {...props} />);
});

describe('ChallengeModes additional settings demo', () => {

  let mockedProps;
  beforeEach(() => {
    mockedProps = getBaseProps()
  });


  test('render additional settings as demo', () => {
    render(<AdditionalSettingsDemo {...mockedProps} />);

    waitFor(() => expect(screen.queryByLabelText('Rules')).toBeTruthy());
  });

  test('render additional settings with a custom board size', () => {
    mockedProps.conf.selected_board_size = "custom";

    render(<AdditionalSettingsDemo {...mockedProps} />);

    expect(screen.queryByTestId('challenge-goban-width')).toBeTruthy();
    expect(screen.queryByTestId('challenge-goban-height')).toBeTruthy();
  });
});
