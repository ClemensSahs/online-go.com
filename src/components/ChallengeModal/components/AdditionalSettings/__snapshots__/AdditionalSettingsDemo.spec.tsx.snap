// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componentSanityChecks for BasicSettings.as demo matches the snapshot 1`] = `
<div
  className="right-pane pane form-horizontal"
  id="challenge-basic-settings"
  role="form"
>
  <div
    className="form-group"
    id="challenge.game.rules-group"
  >
    <label
      className="control-label"
      htmlFor="rules"
    >
      Rules
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <select
          className="challenge-dropdown form-control"
          onChange={[Function]}
        >
          <option
            value="aga"
          >
            AGA
          </option>
          <option
            value="chinese"
          >
            Chinese
          </option>
          <option
            value="ing"
          >
            Ing SST
          </option>
          <option
            value="japanese"
          >
            Japanese
          </option>
          <option
            value="korean"
          >
            Korean
          </option>
          <option
            value="nz"
          >
            New Zealand
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="form-group"
    id="challenge-board-size-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-board-size"
    >
      Board Size
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <select
          className="challenge-dropdown form-control"
          id="challenge-board-size"
          onChange={[Function]}
          value="19x19"
        >
          <optgroup
            label="Normal Sizes"
          >
            <option
              value="19x19"
            >
              19x19
            </option>
            <option
              value="13x13"
            >
              13x13
            </option>
            <option
              value="9x9"
            >
              9x9
            </option>
          </optgroup>
          <optgroup
            label="Extreme Sizes"
          >
            <option
              disabled={false}
              value="25x25"
            >
              25x25
            </option>
            <option
              disabled={false}
              value="21x21"
            >
              21x21
            </option>
            <option
              disabled={false}
              value="5x5"
            >
              5x5
            </option>
          </optgroup>
          <optgroup
            label="Non-Square"
          >
            <option
              disabled={false}
              value="19x9"
            >
              19x9
            </option>
            <option
              disabled={false}
              value="5x13"
            >
              5x13
            </option>
          </optgroup>
          <optgroup
            label="Custom"
          >
            <option
              disabled={false}
              value="custom"
            >
              Custom Size
            </option>
          </optgroup>
        </select>
      </div>
    </div>
  </div>
</div>
`;

exports[`componentSanityChecks for BasicSettings.normal matches the snapshot 1`] = `
<div
  className="right-pane pane form-horizontal"
  id="challenge-basic-settings"
  role="form"
>
  <div
    className="form-group"
    id="challenge.game.rules-group"
  >
    <label
      className="control-label"
      htmlFor="rules"
    >
      Rules
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <select
          className="challenge-dropdown form-control"
          onChange={[Function]}
        >
          <option
            value="aga"
          >
            AGA
          </option>
          <option
            value="chinese"
          >
            Chinese
          </option>
          <option
            value="ing"
          >
            Ing SST
          </option>
          <option
            value="japanese"
          >
            Japanese
          </option>
          <option
            value="korean"
          >
            Korean
          </option>
          <option
            value="nz"
          >
            New Zealand
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="form-group"
    id="challenge-board-size-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-board-size"
    >
      Board Size
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <select
          className="challenge-dropdown form-control"
          id="challenge-board-size"
          onChange={[Function]}
          value="19x19"
        >
          <optgroup
            label="Normal Sizes"
          >
            <option
              value="19x19"
            >
              19x19
            </option>
            <option
              value="13x13"
            >
              13x13
            </option>
            <option
              value="9x9"
            >
              9x9
            </option>
          </optgroup>
          <optgroup
            label="Extreme Sizes"
          >
            <option
              disabled={false}
              value="25x25"
            >
              25x25
            </option>
            <option
              disabled={false}
              value="21x21"
            >
              21x21
            </option>
            <option
              disabled={false}
              value="5x5"
            >
              5x5
            </option>
          </optgroup>
          <optgroup
            label="Non-Square"
          >
            <option
              disabled={false}
              value="19x9"
            >
              19x9
            </option>
            <option
              disabled={false}
              value="5x13"
            >
              5x13
            </option>
          </optgroup>
          <optgroup
            label="Custom"
          >
            <option
              disabled={false}
              value="custom"
            >
              Custom Size
            </option>
          </optgroup>
        </select>
      </div>
    </div>
  </div>
</div>
`;
